A GitHub action told me to try out using the new ``vcpkg-cmake`` port commands.

I did, and for directxtk and directxtk12 they fail because my CMakeLists generates C headers via a custom_command that needs to be used for building:

```
CMake Error at CMakeLists.txt:152 (add_custom_command):
  OUTPUT path

    D:/vcpkg/buildtrees/directxtk/src/sept2021-e237a84202.clean/Src/Shaders/Compiled/SpriteEffect_SpriteVertexShader.inc

  in a source directory as an output of custom command.
```

--- a/ports/directxmesh/portfile.cmake
+++ b/ports/directxmesh/portfile.cmake
@@ -26,14 +26,13 @@ else()
   set(EXTRA_OPTIONS -DBUILD_TOOLS=ON)
 endif()
 
-vcpkg_configure_cmake(
+vcpkg_cmake_configure(
     SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
     OPTIONS ${FEATURE_OPTIONS} ${EXTRA_OPTIONS}
 )
 
-vcpkg_install_cmake()
-vcpkg_fixup_cmake_targets(CONFIG_PATH cmake)
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(CONFIG_PATH cmake)
 
 if((VCPKG_HOST_IS_WINDOWS) AND (VCPKG_TARGET_ARCHITECTURE MATCHES x64))
   vcpkg_download_distfile(

--- a/ports/directxmesh/vcpkg.json
+++ b/ports/directxmesh/vcpkg.json
@@ -8,7 +8,15 @@
   "supports": "windows | linux",
   "dependencies": [
     "directx-headers",
-    "directxmath"
+    "directxmath",
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    }
   ],
   "features": {
     "dx12": {

--- a/ports/uvatlas/portfile.cmake
+++ b/ports/uvatlas/portfile.cmake
@@ -26,14 +26,13 @@ else()
   set(EXTRA_OPTIONS -DBUILD_TOOLS=ON)
 endif()
 
-vcpkg_configure_cmake(
+vcpkg_cmake_configure(
     SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
     OPTIONS ${FEATURE_OPTIONS} ${EXTRA_OPTIONS}
 )
 
-vcpkg_install_cmake()
-vcpkg_fixup_cmake_targets(CONFIG_PATH cmake)
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(CONFIG_PATH cmake)
 
 if((VCPKG_HOST_IS_WINDOWS) AND (VCPKG_TARGET_ARCHITECTURE MATCHES x64) AND (NOT ("eigen" IN_LIST FEATURES)))
   vcpkg_download_distfile(

--- a/ports/uvatlas/vcpkg.json
+++ b/ports/uvatlas/vcpkg.json
@@ -16,6 +16,14 @@
     {
       "name": "directxtex",
       "platform": "!(uwp | linux)"
+    },
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
     }
   ],
   "features": {

--- a/versions/d-/directxmesh.json
+++ b/versions/d-/directxmesh.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "93dd74331b7096440e8dfb6e57c275f87e3cced6",
+      "git-tree": "c6e38f3e2c1f90556658d75ae6edd0e170ab0ac4",
       "version-string": "sept2021",
       "port-version": 0
     },

--- a/versions/u-/uvatlas.json
+++ b/versions/u-/uvatlas.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "85bf7f2c835403d6946ffa6bcd1a7034d7b66c58",
+      "git-tree": "aa1c69c4376ea2d6332065397292b386d60984a4",
       "version-string": "sept2021",
       "port-version": 0
     },

